src/Controller/Plugin/Index.php

```diff
@@ -22,61 +22,16 @@
 
 namespace Shopgate\Base\Controller\Plugin;
 
-use Magento\Framework\App\Action\Action;
-use Magento\Framework\App\Action\Context;
-use Magento\Framework\Module\ResourceInterface;
-use Shopgate\Base\Model\Forwarder;
-use Shopgate\Base\Model\Utility\Registry;
-
 /**
- * Merchant API to Magento2 call router
- *
- * @package Shopgate\Base\Controller\Plugin
+ * Conditional controller for backwards compatibility with PHP 7.1<
+ * logic. Necessary because Magento 2.3 removed PHP 5.6 & 7.0 support.
  */
-class Index extends Action
-{
-    /** @var Forwarder */
-    private $forwarder;
-    /** @var ResourceInterface */
-    private $versionInfo;
-    /** @var Registry */
-    private $registry;
-
-    /**
-     * @param Context           $context
-     * @param Forwarder         $forwarder
-     * @param ResourceInterface $versionInfo
-     * @param Registry          $registry
-     */
-    public function __construct(
-        Context $context,
-        Forwarder $forwarder,
-        ResourceInterface $versionInfo,
-        Registry $registry
-    ) {
-        $this->forwarder   = $forwarder;
-        $this->versionInfo = $versionInfo;
-        $this->registry    = $registry;
-        parent::__construct($context);
+if (PHP_VERSION_ID < 70100) {
```

Comment:
Shouldn't this be just a check for mage >= 2.3 as it implies that php 7.1 is used, whereas the usage of php 7.1 not implies that also mage 2.3 or higher was used. This would result in us using the CsrfAwareActionInterface, but it could be not there.

Different approach, we could just have a file (which is using the Magento\Framework\App\ namespace) that will introduce a blank CsrfAwareActionInterface in case interface_exists('Magento\Framework\App\CsrfAwareActionInterface') fails. So the old Shopgate\Base\Controller\Plugin\ could just implement CsrfAwareActionInterface (if it's mage 2.3 it will use the real interface. If not, it's using the fake one automatically).

Reply:
1) yup you are right
2) this sounds like mage1 shenanigans, I will try it

Reply:
2 is not quite a solution like it would be in mage1 instances as we still need to implement the functions, but the functions themselves are not compatible. Per your suggestion:
```
class Index extends Action implements CsrfAwareActionInterface {
    // incompatible functions need to be here for 2.3>=, but not for 2.3<
}
```

Reply:
Mkay, then just stick with recommendation #1. Hopefully #2 was still amusing.

Reply:
I was rooting for 2, it was a cleaner solution :(

Reply:
Also fine, if you can make it.